fix bug for spacy_upgrade when pip=FALSE

amatsuo opened the pull request:
Close #183 

The issue was: 

- When `conda install` was used internally, conda will not update the package unless the version number is explicitly provided. 

Solution:

- Handing the specific version number when calling an internal function `process_spacy_installation_conda()`, instead of just saying "latest".

kbenoit commented:
Thanks @amatsuo, much appreciated. But this didn't quite work for me. See below.

```
> library(spacyr)
> spacy_upgrade()
checking spaCy version
A new version of spaCy (2.2.3) was found (installed version: 2.1.8)
Do you want to upgrade? 

1: No
2: Yes

Selection: 2
"Yes" was chosen. spaCy will be upgraded.

Do you also want to re-download language model en_core_web_sm? 

1: No
2: Yes

Selection: 2
Using existing conda environment  spacy_condaenv  for spaCy installation
. 
spaCy and language model(s): en_core_web_sm will be installed.  Installing spaCy...
Collecting package metadata (repodata.json): ...working... done
Solving environment: ...working... done

## Package Plan ##

  environment location: /Users/kbenoit/anaconda3/envs/spacy_condaenv

  added / updated specs:
    - python


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    certifi-2019.11.28         |           py36_0         156 KB
    ------------------------------------------------------------
                                           Total:         156 KB

The following packages will be UPDATED:

  ca-certificates    conda-forge::ca-certificates-2019.11.~ --> pkgs/main::ca-certificates-2020.1.1-0
  openssl            conda-forge::openssl-1.1.1d-h0b31af3_0 --> pkgs/main::openssl-1.1.1d-h1de35cc_4

The following packages will be SUPERSEDED by a higher-priority channel:

  certifi                                       conda-forge --> pkgs/main



Downloading and Extracting Packages
certifi-2019.11.28   | 156 KB    | ########## | 100% 
Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... done


==> WARNING: A newer version of conda exists. <==
  current version: 4.7.10
  latest version: 4.8.2

Please update conda by running

    $ conda update -n base -c defaults conda


Collecting package metadata (repodata.json): ...working... done
Solving environment: ...working... done

## Package Plan ##

  environment location: /Users/kbenoit/anaconda3/envs/spacy_condaenv

  added / updated specs:
    - python
    - spacy=2.2.3


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    catalogue-1.0.0            |             py_0          10 KB  conda-forge
    certifi-2019.11.28         |           py36_0         148 KB  conda-forge
    cython-blis-0.4.1          |   py36h0b31af3_0         3.2 MB  conda-forge
    libcxx-9.0.1               |                1        1011 KB  conda-forge
    preshed-3.0.2              |   py36h4a8c4bd_1          94 KB  conda-forge
    spacy-2.2.3                |   py36ha1b3eb9_0         8.6 MB  conda-forge
    srsly-1.0.0                |   py36h4a8c4bd_0         196 KB  conda-forge
    thinc-7.3.0                |   py36ha1b3eb9_0         1.6 MB  conda-forge
    ------------------------------------------------------------
                                           Total:        14.8 MB

The following NEW packages will be INSTALLED:

  catalogue          conda-forge/noarch::catalogue-1.0.0-py_0
  importlib_metadata conda-forge/osx-64::importlib_metadata-1.5.0-py36_0
  zipp               conda-forge/noarch::zipp-3.0.0-py_0

The following packages will be UPDATED:

  cython-blis                          0.2.4-py36h01d97ff_0 --> 0.4.1-py36h0b31af3_0
  libcxx                 pkgs/main::libcxx-4.0.1-hcfea43d_1 --> conda-forge::libcxx-9.0.1-1
  numpy                               1.16.3-py36hdf140aa_0 --> 1.18.1-py36hde6bac1_0
  preshed            pkgs/main::preshed-2.0.1-py36h0a44026~ --> conda-forge::preshed-3.0.2-py36h4a8c4bd_1
  spacy                                2.1.8-py36h770b8ee_0 --> 2.2.3-py36ha1b3eb9_0
  srsly                                0.0.5-py36h0a44026_0 --> 1.0.0-py36h4a8c4bd_0
  thinc                                7.0.4-py36h770b8ee_0 --> 7.3.0-py36ha1b3eb9_0

The following packages will be SUPERSEDED by a higher-priority channel:

  ca-certificates     pkgs/main::ca-certificates-2020.1.1-0 --> conda-forge::ca-certificates-2019.11.28-hecc5488_0
  certifi                                         pkgs/main --> conda-forge
  openssl              pkgs/main::openssl-1.1.1d-h1de35cc_4 --> conda-forge::openssl-1.1.1d-h0b31af3_0



Downloading and Extracting Packages
preshed-3.0.2        | 94 KB     | ########## | 100% 
srsly-1.0.0          | 196 KB    | ########## | 100% 
libcxx-9.0.1         | 1011 KB   | ########## | 100% 
certifi-2019.11.28   | 148 KB    | ########## | 100% 
catalogue-1.0.0      | 10 KB     | ########## | 100% 
spacy-2.2.3          | 8.6 MB    | ########## | 100% 
cython-blis-0.4.1    | 3.2 MB    | ########## | 100% 
thinc-7.3.0          | 1.6 MB    | ########## | 100% 
Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... done


==> WARNING: A newer version of conda exists. <==
  current version: 4.7.10
  latest version: 4.8.2

Please update conda by running

    $ conda update -n base -c defaults conda


Installing model "en_core_web_sm"

Traceback (most recent call last):
  File "/Users/kbenoit/anaconda3/envs/spacy_condaenv/lib/python3.6/runpy.py", line 183, in _run_module_as_main
    mod_name, mod_spec, code = _get_module_details(mod_name, _Error)
  File "/Users/kbenoit/anaconda3/envs/spacy_condaenv/lib/python3.6/runpy.py", line 142, in _get_module_details
    return _get_module_details(pkg_main_name, error)
  File "/Users/kbenoit/anaconda3/envs/spacy_condaenv/lib/python3.6/runpy.py", line 109, in _get_module_details
    __import__(pkg_name)
  File "/Users/kbenoit/anaconda3/envs/spacy_condaenv/lib/python3.6/site-packages/spacy/__init__.py", line 13, in <module>
    from .cli.info import info as cli_info
  File "/Users/kbenoit/anaconda3/envs/spacy_condaenv/lib/python3.6/site-packages/spacy/cli/__init__.py", line 1, in <module>
    from .download import download  # noqa: F401
  File "/Users/kbenoit/anaconda3/envs/spacy_condaenv/lib/python3.6/site-packages/spacy/cli/download.py", line 9, in <module>
    from wasabi import msg
ImportError: cannot import name 'msg'
Error: Error 1 occurred installing packages into conda environment spacy_condaenv
> spacy_upgrade()
checking spaCy version
Your spaCy version is the latest available.
> spacy_initialize()
spacy python option is already set, spacyr will use:
	condaenv = "spacy_condaenv"
 Error in py_run_file_impl(file, local, convert) : 
  ImportError: cannot import name 'msg'
```

kbenoit commented:
I got it working after `spacy_uninstall()`, then `spacy_install()`, but now on macOS having the same problem as on Windows, where the "en" alias to the language model is not being set. Specifying the full language model name works though.

```r
> spacy_parse(data_char_paragraph)
spacy python option is already set, spacyr will use:
	condaenv = "spacy_condaenv"
 Error in py_run_file_impl(file, local, convert) : 
  OSError: [E050] Can't find model 'en'. It doesn't seem to be a shortcut link, a Python package or a valid path to a data directory. 
> spacy_finalize()
Error in spacy_finalize() : Nothing to finalize. spaCy is not initialized
> spacy_initialize(model = "en_core_web_sm")
Python space is already attached.  If you want to switch to a different Python, please restart R.
successfully initialized (spaCy Version: 2.1.8, language model: en_core_web_sm)
(python options: type = "condaenv", value = "spacy_condaenv")
> spacy_parse(data_char_paragraph)
    doc_id sentence_id token_id        token        lemma   pos     entity
1    text1           1        1      Instead      instead   ADV           
2    text1           1        2           we       -PRON-  PRON  
```

amatsuo commented:
For the question about the upgrade, you need to restart R session. 

I added the message for it:
```
Successfully upgraded
Condaenv: spacy_test; Langage model(s): en_core_web_sm
For the upgrade to take effect, please restart R session
```

Also, for the model name shortcut, this is a bug (which is fixed `issue-151`, we need to do something with the word vector incorporation...). I changed the default model name from `en` to `en_core_web_sm`.

kbenoit commented:
Should the install script also use a newer version of conda? I see
```
==> WARNING: A newer version of conda exists. <==
  current version: 4.7.10
  latest version: 4.8.2
```
